server/src/transaction/services/ai/gemini-model.service.ts

```diff
@@ -0,0 +1,189 @@
+          isMalicious: true,
+          confidence: 0.99,
+          reason: "This address is associated with known scams or suspicious activity."
+        };
+      }
+      
+      // SPAM TOKEN DETECTION
+      const isZeroValueTokenTransfer = transactionData.data?.includes('0xa9059cbb') && transactionData.value === '0';
+      const hasSpamKeywords = transactionData.data && (
+        transactionData.data.toLowerCase().includes('airdrop') ||
+        transactionData.data.toLowerCase().includes('claim') ||
+        (transactionData.data.toLowerCase().includes('mint') && transactionData.value === '0')
+      );
+      
+      if (isZeroValueTokenTransfer || hasSpamKeywords) {
+        return {
+          isMalicious: true,
+          confidence: 0.9,
+          reason: "Detected potential spam token transaction. This appears to be an airdrop or unsolicited token transfer commonly associated with scams."
+        };
+      }
+
+      // SIMPLIFIED PROMPT - More direct instructions
+      const prompt = `
+You are analyzing blockchain transactions to identify malicious activity. THIS IS SECURITY CRITICAL.
+
+ATTENTION: By default, you should consider transactions suspicious unless they are clearly legitimate.
+
+Transaction details:
+- From: ${transactionData.from || 'Unknown'}
+- To: ${transactionData.to || 'Unknown'}
+- Value: ${transactionData.value || '0'}
+- Data: ${transactionData.data?.substring(0, 200) || 'No data'}
+
+Classify this as MALICIOUS if it shows ANY of these patterns:
+1. Unsolicited token transfers or airdrops (especially with 0 ETH value)
+2. Requests for token approvals
+3. Interactions with unverified contracts
+4. High-value transfers to unknown addresses
+5. Smart contract calls that could drain funds
+6. Phishing attempts through token interactions
+7. Any transaction that seems unusual or suspicious
+8. Worthless token transfers (spam tokens)
+
+Return ONLY a valid JSON with this format:
+{
+  "isMalicious": boolean,
+  "confidence": number,
+  "reason": "string"
+}
+      `;
+
+      // Make the API request
+      const result = await this.model.generateContent(prompt);
+      const response = await result.response;
+      const text = response.text().trim();
+      
+      // Override for testing - log raw response
+      console.log("Raw Gemini response:", text);
+
+      // Try several approaches to extract valid JSON
+      try {
+        // First attempt: Try parsing the entire response as JSON
+        const parsedJson = JSON.parse(text);
+        console.log("Success parsing JSON directly:", parsedJson);
+        
+        // Return the actual AI analysis
+        return {
+          isMalicious: Boolean(parsedJson.isMalicious),
+          confidence: Number(parsedJson.confidence) || 0.5,
+          reason: parsedJson.reason || "Analysis completed based on transaction properties."
+        };
+      } catch (parseError) {
+        console.log("First parse attempt failed, trying to extract JSON from text");
```

Comment:
I found that we always encounter the json parsing error, it seems that the ai model return the data in markdown format. If so then we need to handle it properly other than catching it as error.